### DISCUSSION TITLE
Amelia output with fixed effects regression

### DISCUSSION THREAD

**Discussion by flxflks** [IQSS/Amelia#53] [Q&A]
I used Amelia to impute the missing data in my time-series cross-sectional data. However, when reading the vignette I couldn't find any example to use Amelia with fixed / random regressions. Notably, `plm` and `fixest` do not work with `with` for me. Is there a package that allows to model fixed effect regressions with an Amelia object?

**Comment by flxflks** [comment]
Crossposted on [SO](https://stackoverflow.com/questions/78123287/fixed-effects-regressions-with-amelia-object)

**Comment by mattblackwell** [comment]
Yeah, the `with()` function for Amelia only works with models that don't need the data argument. For situations like this, you can use `lapply()` to do the fits like so:
```{r}
library(tidyverse)
library(broom)
library(Amelia)
library(fixest)
library(plm)

# Load dataset
data("africa", package = "Amelia")

# Multiple imputation
a.out <- amelia(x = africa, m = 10, cs = "country",
                ts = "year", logs = "gdp_pc", p2s = 0)

# Run fixed effects regression
fits <- lapply(
  a.out$imputations,
  \(x) feols(gdp_pc ~ infl * trade, data = x)
)

mi.combine(fits)
```
